eng/service.proj

```diff
@@ -10,8 +10,10 @@
     <IncludeStress Condition="'$(IncludeStress)' == ''">true</IncludeStress>
     <IncludeSamplesApplications Condition="'$(IncludeSamplesApplications)' == ''">true</IncludeSamplesApplications>
     <IncludeSamplesApplications Condition="'$(ServiceDirectory)' != '*' or '$(IncludeSamples)' == 'false'">false</IncludeSamplesApplications>
+
+    <CodeCoverageDirectory>$([System.IO.Path]::GetFullPath("$(CodeCoverageDirectory)", "$(MSBuildThisFileDirectory)..\sdk"))</CodeCoverageDirectory>
     <TraversalGlobalProperties>
-      CodeCoverageDirectory=$([System.IO.Path]::GetFullPath("$(CodeCoverageDirectory)", "$(MSBuildThisFileDirectory)..\sdk"));
+      CodeCoverageDirectory=$(CodeCoverageDirectory);
```

Comment:
Having different value between CodeCoverageDirectory in different msbuild invocations results in two builds per dll.

Reply:
I agree that having different global properties causes different builds however I don't understand how your change here actually fixes anything.

Reply:
ResolveReferences step doesn't pass the `TraversalGlobalProperties` but still inherits the current value.

![image](https://user-images.githubusercontent.com/1697911/115930570-90d8ff80-a43e-11eb-86d6-2b68c43e1520.png)

But build target passes `CodeCoverageDirectory` so we get a different set of global variables:

![image](https://user-images.githubusercontent.com/1697911/115930605-a0f0df00-a43e-11eb-82c0-f1078a35b5dc.png)

My change makes them the same in both contexts.